### PR TITLE
Add option to query cached results

### PR DESCRIPTION
**Pull request by yoavbls**
#2285 
I Implemented it exactly like Arik's example in the issue 😃 
image

### CODE REVIEW

File: redash/query_runner/query_results.py [hunk 1]
```diff
@@ -60,21 +65,29 @@ def _load_query(user, query_id):
     return query
 
 
-def create_tables_from_query_ids(user, connection, query_ids):
-    for query_id in set(query_ids):
+def get_query_results(user, query_id, bring_from_cache):
         query = _load_query(user, query_id)
+        if bring_from_cache and query.latest_query_data_id is not None:
```
Comment (arikfr) [inline redash/query_runner/query_results.py]: When there are no results stored and the user asked for cache we shouldn't fallback to running the query but rather show an error message saying this ("No cached result available for query xxx.").

Comment (arikfr) [reply redash/query_runner/query_results.py]: Also seems like the indentation here is a bit off.

File: redash/query_runner/query_results.py [hunk 2]
```diff
@@ -60,21 +65,29 @@ def _load_query(user, query_id):
...
-                "Failed loading results for query id {}.".format(query.id))
-
-        results = json.loads(results)
+def create_tables_from_query_ids(user, connection, query_ids, cached_query_ids=[]):
+    for query_id in set(query_ids):
+        results = get_query_results(user, query_id, False)
         table_name = 'query_{query_id}'.format(query_id=query_id)
         create_table(connection, table_name, results)
 
+    for query_id in set(cached_query_ids):
```
Comment (arikfr) [inline redash/query_runner/query_results.py]: It's worth loading the cached results first.

File: redash/query_runner/query_results.py (hunk 1 above)
Comment (yoavbls) [reply redash/query_runner/query_results.py]: Sounds right, i fixed it

File: redash/query_runner/query_results.py (hunk 2 above)
Comment (yoavbls) [reply redash/query_runner/query_results.py]: I hope you mean to run this for loop first, 
That's what I changed in my last commit

**Comment by yoavbls** [comment]
Thank you for the code review,
I hope its fine now :)
(and sorry for the reference from another pull request, I fixed it but its still appears)

**Comment by arikfr** [comment]
Thanks!

**Pull request merged**